File: Difficulty/Medium/[48] Rotate Image/solution.py
class Solution:
    def rotate(self, matrix: List[List[int]]) -> None:
        """
        Do not return anything, modify matrix in-place instead.
        """
        n, begin, end = len(matrix) - 1, 0, len(matrix) - 1
        while begin < end:
            for i in range(begin, end):
                matrix[begin][i], matrix[i][n-begin] = matrix[i][n-begin], matrix[begin][i]
                matrix[begin][i], matrix[n-begin][n-i] = matrix[n-begin][n-i], matrix[begin][i]
                matrix[begin][i], matrix[n-i][begin] = matrix[n-i][begin], matrix[begin][i]
            begin += 1
            end -= 1
        

File: Difficulty/Medium/[48] Rotate Image/solution.cpp
/*
 * @lc app=leetcode id=48 lang=cpp
 *
 * [48] Rotate Image
 */
class Solution {
public:
    void rotate(vector<vector<int>>& matrix) {
        int n = matrix.size() - 1;
        int begin = 0;
        int end = n;
        while (begin < end){
            for (int i = begin; i < end; i++){
                swap(matrix[begin][i], matrix[i][n-begin]);
                swap(matrix[begin][i], matrix[n-begin][n-i]);
                swap(matrix[begin][i], matrix[n-i][begin]);
            }
            begin++;
            end--;
        }
    }
};

